packages/client/src/runtime/core/model/applyFluent.ts

```diff
@@ -119,7 +119,7 @@ export function applyFluent(
         // we allow for chaining more with this recursive call
         return applyFluent(client, ...modelArgs, ...dataArgs)
       },
-      ...defaultProxyHandlers(ownKeys),
+      ...defaultProxyHandlers([...ownKeys, ...Object.getOwnPropertyNames(prismaPromise)]),
```

Comment:
This currently just grabs own properties from prismaPromise which works fine since prismaPromises are created as an object with methods defined inline.  If prismaPromise were converted to a class in the future, properties like `then` would move to the prototype, and this would break.

The way "ownKeys" are used in the proxy is a little weird, and acts basically like flattening the prototype chain so all props are treated as own props. 

I think the technically more correct solution (not worth it in my opinion) would be to walk up the prototype chain and get the set of all ownPropertyNames and ownPropertySymbols.  I'm not sure that actually adds any real world value though, so going to leave it as is, but happy to make this more robust if you want.